### PR TITLE
Assemble bilinear forms on FaceDomain

### PR DESCRIPTION
**Pull request by ghislainb**

### CODE REVIEW

File: src/LazyOperators/lazy_operator.jl [hunk 1]
```diff
@@ -9,9 +9,9 @@ and optionally:
 """
 abstract type AbstractLazy end
 
-function materialize(a::AbstractLazy, x)
-    error("`materialize` is not defined for:\n $(typeof(a)) \n and:\n $(typeof(x))")
-end
+# function materialize(a::AbstractLazy, x)
```
Comment (bmxam) [inline src/LazyOperators/lazy_operator.jl:12]: can you remove it if it's not necessary anymore?

File: src/algebra/gradient.jl [hunk 2]
```diff
@@ -202,3 +210,29 @@ function LazyOperators.materialize(
...
+    lOp::Gradient{O, <:Tuple{LazyMapOver}},
+    sidePoint::AbstractSide{Nothing, <:Tuple{FacePoint}},
+) where {O}
+    ∇x = tuplemap(x -> materialize(∇(x), sidePoint), get_args(get_args(lOp)...))
+    return MapOver(∇x)
+end
+
+function LazyOperators.materialize(
+    lOp::Gradient{O, <:Tuple{Vararg{AbstractCellFunction}}},
+    side::AbstractSide{Nothing, <:Tuple{FacePoint}},
```
Comment (bmxam) [inline src/algebra/gradient.jl:233]: for coherence, can you name the argument `sidePoint`?

File: test/dof/test_assembler.jl [hunk 3]
```diff
@@ -210,6 +210,72 @@
...
+
+        uref = PhysicalFunction(x -> 3 * x[1] + x[2]^2 + 2 * x[1]^3)
+        f = PhysicalFunction(x -> -2 - 12 * x[1])
+        g = uref # boundary condition
+
+        avg(u) = 0.5 * (side⁺(u) + side⁻(u))
+
+        # Build mesh
+        meshParam = (nx = n + 1, ny = n + 1, lx = Lx, ly = Lx, xc = 0.0, yc = 0.0)
+        tmp_path = "tmp.msh"
```
Comment (bmxam) [inline test/dof/test_assembler.jl:229]: tiny improvement 
`tmp_path = joinpath(mktemp(), "mesh.msh")` or `tmp_path = joinpath(tempdir(), "mesh.msh")` and then you don't have to remove the file after reading it

**Review by bmxam — DISMISSED** [review]
just tiny improvements, then you can merge without my re-approval

can you also increase the version number of Bcube please? This will help tracking compat of tutorials

File: src/algebra/gradient.jl (hunk 2 above)
Comment (ghislainb) [reply src/algebra/gradient.jl]: done

File: src/LazyOperators/lazy_operator.jl (hunk 1 above)
Comment (ghislainb) [reply src/LazyOperators/lazy_operator.jl]: done

File: test/dof/test_assembler.jl (hunk 3 above)
Comment (ghislainb) [reply test/dof/test_assembler.jl:229]: It does not work on my local computer. Let's do this in a next PR.

**Review by bmxam — APPROVED** [review]

**Pull request merged**